wsmaster/codenvy-hosted-api-organization/src/main/java/com/codenvy/organization/api/resource/OrganizationalAccountAvailableResourcesProvider.java

```diff
@@ -0,0 +1,184 @@
+        this.resourceUsageManagerProvider = resourceUsageManagerProvider;
+        this.resourceAggregator = resourceAggregator;
+        this.organizationManager = organizationManager;
+    }
+
+    @Override
+    public List<? extends Resource> getAvailableResources(String accountId) throws NotFoundException, ServerException {
+        Organization organization = organizationManager.getById(accountId);
+
+        if (organization.getParent() == null) {
+            return getAvailableOrganizationResources(organization);
+        } else {
+            return min(getAvailableOrganizationResources(organizationManager.getById(organization.getParent())),
+                       getAvailableOrganizationResources(organization));
+        }
+    }
+
+    /**
+     * Returns total resources minus resources which are already
+     * used by organization or by any of its suborganizations.
+     *
+     * @param organization
+     *         organization id to calculate its available resources
+     * @return resources which are available for usage by specified organization
+     * @throws NotFoundException
+     *         when organization with specified id doesn't exist
+     * @throws ServerException
+     *         when any other exception occurs on calculation of available resources
+     */
+    private List<? extends Resource> getAvailableOrganizationResources(Organization organization) throws NotFoundException,
+                                                                                                         ServerException {
+        final ResourceUsageManager resourceUsageManager = resourceUsageManagerProvider.get();
+        final List<? extends Resource> total = resourceUsageManager.getTotalResources(organization.getId());
+        final List<Resource> unavailable = new ArrayList<>(resourceUsageManager.getUsedResources(organization.getId()));
+        unavailable.addAll(getUsedResourcesBySuborganizations(organization.getQualifiedName()));
+        try {
+            return resourceAggregator.deduct(total, unavailable);
+        } catch (NoEnoughResourcesException e) {
+            LOG.warn("Organization with id {} uses more resources {} than it has {}.", organization.getId(), format(unavailable),
+                     format(total));
+            return resourceAggregator.excess(total, unavailable);
+        }
+    }
+
+    /**
+     * Returns list that contains resources with minimum amounts.
+     *
+     * <p>Note that if one of the lists don't contain resource
+     * than resource also won't contain it.
+     *
+     * @param resources1
+     *         the fist resources list to compare
+     * @param resources2
+     *         the second resources list to compare
+     * @return resources with minimum amounts.
+     */
+    private List<Resource> min(Collection<? extends Resource> resources1, Collection<? extends Resource> resources2) {
```

Comment:
Does it make sense to separate this method on 2, 
1 is intersect, the second is aggregation by minimal amount? 
I spend some time getting what the method is doing.